commands/economy/buy.js

```diff
@@ -15,21 +15,14 @@ module.exports = {
   executeInteraction (client, locale, interaction) {
     if (interaction.database.economyEnabled !== 0) {
       if (interaction.options.getString('productname')) {
-        interaction.guild.locale = locale
-        fetchShopProduct(client, interaction.guild, interaction.options.getString('productname'), (productData) => {
-          if (productData) {
-            if (interaction.options.getString('properties')) productData.userInput = interaction.options.getString('properties').split(',')
-            buyItem(client, interaction.member, interaction.guild, productData, (status) => {
-              if (status.code) {
-                interaction.editReply({ embeds: [Success(status.ia || i18n(locale, 'BUY::SUCCESS', { ITEM: productData.productName }))] })
-              } else {
-                interaction.editReply({ embeds: [Error(status.ia || i18n(locale, 'BUY::NOMONEY', { ITEM: productData.productName }))] })
-              }
-            })
-          } else {
-            interaction.editReply({ embeds: [Error(i18n(locale, 'BUY::NOTFOUND', { ITEM: interaction.options.getString('productname') }))] })
-          }
-        })
+        try {
+          buyItem(client, interaction.member, interaction.guild, interaction.options.getString('productname'), interaction.options.getString('properties'), (err) => {
+            if (err.message === 'ECO_200') interaction.editReply({ embeds: [Success(i18n(locale, 'BUY::SUCCESS', { ITEM: interaction.options.getString('productName') }))] })
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/alexandrumuntenas/pingu/pullRequest?prid=8657799)
```suggestion
            if (err.message === 'ECO_200') {interaction.editReply({ embeds: [Success(i18n(locale, 'BUY::SUCCESS', { ITEM: interaction.options.getString('productName') }))] })}
```